src/main/java/org/opentest4j/FileInfo.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2015-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.opentest4j;
+
+import java.io.Serializable;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+
+/**
+ * {@code FileInfo} is a pair of a file path and the contents of the file.
+ * <p>
+ * {@code FileInfo} is expected to be used as {@code actual} and {@code expected} values in {@code AssertionFailedError}
+ * <p>
+ * The semantics of the {@code path} can be defined by the users of this class.
+ *
+ * @author Marc Philipp
+ * @author Reinhold Degenfellner
+ * @since 1.2.1
+ */
+public class FileInfo implements Serializable {
+
+	private static final long serialVersionUID = 1L;
+
+	private final String path;
+	private final byte[] contents;
+
+	/**
+	 * Constructs a {@code FileInfo} with a path to a file and the contents of the file.
+	 * <p>
+	 * Be cautious with large files (maybe limiting the size is a good idea)
+	 *
+	 * @param path the path to the file; must not be {@code null}
+	 * @param contents the contents of the file; must not be {@code null};
+	 */
+	public FileInfo(String path, byte[] contents) {
```

Comment:
It occurs to me that the `contents` should be copied when passed into the constructor to prevent users from changing them afterwards and to prevent thread-safety bugs.

Alternatively, using `String` or a custom immutable `ByteString` type would prevent these issues altogether.

What do you think?

Reply:
My preference would be `ByteString` so that people can't get the ordering of `path` and `contents` mixed up, or to turn this constructor into a builder.

Reply:
That being said, taking protobuf's `ByteString` class as an example, its API is quite complex, so maybe copying the byte array or making it a `String` (making sure to turn the constructor into a builder) would be preferable. 🤔

Reply:
I feel like in this case avoiding a copy can be a bonus. since the input files might be huge.
I don't think we want to use more 3rd party libraries for this.
If that turns out to be a problem it is relatively save to introduce that later.

Reply:
Okay! In that case, I'd love to see a comment in the class javadocs that say this class is not immutable. Otherwise I'm reasonably happy. :)